backend/clubs/views.py

```diff
@@ -4395,6 +4396,26 @@ def question_response(self, *args, **kwargs):
             ApplicationQuestion.objects.filter(pk=questions[0]).first().application
         )
         committee = application.committees.filter(name=committee_name).first()
+
+        committees_applied = (
+            ApplicationSubmission.objects.filter(
+                user=self.request.user,
+                committee__isnull=False,
+                application=application,
+                archived=False,
+            )
+            .values_list("committee__name", flat=True)
+            .distinct()
+        )
+
+        # limit applicants to 2 committees
+        if (
+            committee
+            and committees_applied.count() >= 2
+            and committee_name not in committees_applied
+        ):
+            return Response([], status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Yeah that's fair, I'll change this.

> This also saves us from the edge case of "what if there was a random bit flip or something and I got a HTTP 400 but the client side interprets it as this particular error of having already applied to too many committees" -- that's unlikely to happen but still maybe possible.

Curious, can this even happen? I was under the impression that 400s are more deterministic and 500s are the "something's fucked rip" error.

Reply:
5xx is when the server fucks up, 4xx are when the clients fuck up (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status). Generally, 400 means that the server couldn't actually read the request because it was somehow malformed. For the most part 400s won't ever really happen because modern web browsers won't send busted HTTP requests. It's kind of abusing notation for us to manually throw a 400 even though we were able to read the HTTP request.

Also, with a HTTP server like Django, manually throwing a 400 is almost always bad practice because Django will throw the 400 for you automatically if it cannot parse the HTTP request into the Python abstraction we use to interact with the request within Django. Throwing something like a `414 URI Too Long` would be okay though.

Kinda funny when I was looking up HTTP error codes there's also `418 I'm a teapot` for when "The server refuses the attempt to brew coffee with a teapot." lmao

Reply:
Thanks that makes sense HTTP king